editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/index.tsx

```diff
@@ -50,72 +85,30 @@ const Search: React.FC = () => {
       debounce((pattern: string) => {
         console.debug("Search term: ", pattern);
         search(pattern);
-        setLastSearchedTerm(pattern);
+        setLastPattern(pattern);
         setIsSearching(false);
       }, DEBOUNCE_MS),
     [search],
   );
 
-  useEffect(() => {
-    if (formik.values.pattern !== lastSearchedTerm) {
-      setIsSearching(true);
-    }
-  }, [formik.values.pattern, lastSearchedTerm]);
-
   return (
-    <Container component={Box} p={3}>
-      <form onSubmit={formik.handleSubmit}>
-        <Typography
-          component={"label"}
-          htmlFor="pattern"
-          variant="h3"
-          mb={1}
-          display={"block"}
-        >
-          Search this flow and internal portals
-        </Typography>
-        <Box
-          sx={{ display: "flex", position: "relative", alignItems: "center" }}
-        >
-          <Input
-            id="pattern"
-            name="pattern"
-            value={formik.values.pattern}
-            onChange={(e) => {
-              formik.setFieldValue("pattern", e.target.value);
-              formik.handleSubmit();
-            }}
-            inputProps={{ spellCheck: false }}
-          />
-          {isSearching && (
-            <CircularProgress
-              size={25}
-              sx={(theme) => ({
-                position: "absolute",
-                right: theme.spacing(1.5),
-                zIndex: 1,
-              })}
-            />
-          )}
-        </Box>
-        <ChecklistItem
-          label="Search only data fields"
-          id={"search-data-field-facet"}
-          checked
-          inputProps={{ disabled: true }}
-          onChange={() => {}}
-          variant="compact"
-        />
-        <Box pt={3}>
-          {formik.values.pattern && (
-            <>
-              <NodeSearchResults results={results} />
-              <ExternalPortalList />
-            </>
-          )}
-        </Box>
-      </form>
-    </Container>
+    <Virtuoso<Data, Context>
+      totalCount={results.length}
+      context={{
+        results,
+        isSearching,
+        lastPattern,
+        formik,
+        setIsSearching,
+      }}
+      components={{
+        Footer: ExternalPortalList,
+        List: ListComponent,
+        Item: ListItemComponent,
+        Header: SearchHeader,
```

Comment:
Child components are passed in via the `components` prop ([docs](https://virtuoso.dev/footer/)).